src/gui/ar/draw.js

```diff
@@ -2245,33 +2245,26 @@ realityEditor.gui.ar.draw.removeFullscreenFromFrame = function(objectKey, frameK
                 globalDOMCache['iframe' + frame.uuid].classList.remove('envelopeFadingIn');
             }, 1000);
         }, 50);
-        
-        // create a temporary, semi-transparent DOM element in , which then animates to minimize down to the same transform as the frame
-        var envelopeAnimationDiv = document.createElement('div');
-        envelopeAnimationDiv.classList.add('main', 'envelopeAnimationDiv', 'ignorePointerEvents');
-        envelopeAnimationDiv.style.width = globalDOMCache['object' + frame.uuid].style.width;
-        envelopeAnimationDiv.style.height = globalDOMCache['object' + frame.uuid].style.height;
-        // start with a hard-coded MVP matrix that covers the full screen
-        envelopeAnimationDiv.style.transform = "matrix3d(284.7391935492032, 3.070340532377773, 0.0038200291675306924, 0.003834921258919453, -3.141247565648438, 284.35804025980104, 0.011905637861498192, 0.011900616291666024, 20.568534190244556, 9.715687705148639, -0.6879540871592961, -0.6869158438452686, -1268.420885449479, 86.38923398120664, 100200, 260.67004803237324)";
-        envelopeAnimationDiv.style.opacity = 0.5;
-        document.getElementById('GUI').appendChild(envelopeAnimationDiv);
-
-        // wait a small delay so the transition CSS property applies
-        envelopeAnimationDiv.classList.add('animateAllProperties500ms');
-        setTimeout(function() {
-            // animate the transformation matrix
-            var destinationMatrix = globalDOMCache['object' + frame.uuid].style.webkitTransform.split('(')[1].split(')')[0].split(',').map(function(val) {
-                return parseFloat(val);
-            });
-            destinationMatrix[0] *= 0.5; // scale it down a bit so it minimizes even smaller than the non-fullscreen frame
-            destinationMatrix[5] *= 0.5;
-            envelopeAnimationDiv.style.transform = 'matrix3d(' + destinationMatrix.toString() + ')';
-            
-            // also fade it out and remove it entirely when done
-            envelopeAnimationDiv.style.opacity = 0;
-            setTimeout(function() {
-                envelopeAnimationDiv.parentElement.removeChild(envelopeAnimationDiv);
-            }, 250);
+
+        const parentDiv = globalDOMCache['object' + frame.uuid];
+        let tempAnimDiv = document.createElement('div');
+        tempAnimDiv.classList.add('temp-anim-div');
+        tempAnimDiv.style.transform = 'matrix3d(643.374, -0.373505, 0.000212662, 0.000212647, 0.372554, 643.38, 0.000554764, 0.000554727, -2.77404, 4.28636, 0.500033, 0.5, -1406.67, 2173.54, 34481.6, 253.541)';
```

Comment:
Can you add a comment explaining how you got this hard-coded matrix, in case someone else wants to modify it in the future?

Reply:
I added a tool, closed it to reveal the icon, and moved the camera towards the tool, so that the tool almost fills up the screen in the center. And then I get the matrix3d of the object that the tool is attached to. Very hacky, hope to make it procedural in the future😆.